ckanext/csrf_filter/plugin.py

```diff
@@ -95,6 +96,9 @@ def check_csrf():
         def set_csrf_token(response):
             """ Apply a CSRF token to all response bodies.
             """
+            if isinstance(getattr(response, 'response', None), GeneratorType):
```

Comment:
Looks good but there should probably be a comment for future maintainers, explaining why generators are excluded.

Reply:
Comments in now